src/region.js

```diff
@@ -33,155 +33,108 @@ var Region = MarionetteObject.extend({
+      return this;
+    }
 
-      // The views that we're about to attach to the document
-      // It's important that we prevent _getNestedViews from being executed unnecessarily
-      // as it's a potentially-slow method
-      var displayedViews = [];
+    this.triggerMethod('before:show', view, this, options);
 
-      var attachOptions = _.extend({
-        triggerBeforeAttach: this.triggerBeforeAttach,
-        triggerAttach: this.triggerAttach
-      }, showOptions);
+    // We need to listen for if a view is destroyed in a way other than through the region.
+    // If this happens we need to remove the reference to the currentView since once a view
+    // has been destroyed we can not reuse it.
+    view.once('destroy', this.empty, this);
 
-      if (attachedRegion && attachOptions.triggerBeforeAttach) {
-        displayedViews = this._displayedViews(view);
-        this._triggerAttach(displayedViews, 'before:');
-      }
+    // Make this region the view's parent.
+    // It's important that this parent binding happens before rendering so that any events
+    // the child may trigger during render can also be triggered on the child's ancestor views.
+    view._parent = this;
 
-      this.attachHtml(view, _shouldReplaceElement);
-      this.currentView = view;
+    this._renderView(view);
 
-      if (attachedRegion && attachOptions.triggerAttach) {
-        displayedViews = this._displayedViews(view);
-        this._triggerAttach(displayedViews);
-      }
+    if (isChangingView) {
+      this.triggerMethod('before:swapIn', view, this, options);
+    }
 
-      if (isChangingView) {
-        this.triggerMethod('swapOut', changingView, this, options);
-        this.triggerMethod('swapIn', view, this, options);
-      }
+    this._attachView(view, options);
 
-      this.triggerMethod('show', view, this, options);
-      triggerMethodOn(view, 'show', view, this, options);
+    if (isChangingView) {
+      this.triggerMethod('swapOut', changingView, this, options);
+      this.triggerMethod('swapIn', view, this, options);
     }
 
+    this.triggerMethod('show', view, this, options);
     return this;
   },
 
-  shouldDestroyView: function(view, options) {
-    var showOptions     = options || {};
-    var isDifferentView = view !== this.currentView;
-    var preventDestroy  = !!showOptions.preventDestroy;
-
-    return isDifferentView && !preventDestroy;
-  },
-
-  _renderView: function(view, options) {
+  _renderView: function(view) {
+    if (view._isRendered) {
+      return;
+    }
     if (!view.supportsRenderLifecycle) {
       triggerMethodOn(view, 'before:render', view);
     }
-    this.renderView(view, options);
+    view.render();
     if (!view.supportsRenderLifecycle) {
       triggerMethodOn(view, 'render', view);
     }
   },
 
-  renderView: function(view, options) {
-    view.render();
-  },
-
-  triggerBeforeAttach: true,
-  triggerAttach: true,
+  _attachView: function(view, options={}) {
```

Comment:
I'll make a new issue, because given that `isRendered()` means the above, Region code should be fine as it stands.